File: pc_app/backend/app/schemas/route_segment.py
from pydantic import BaseModel


class RouteSegment(BaseModel):
    segment_id: str
    from_node: str
    to_node: str
    label: str
    is_enabled: bool = True
    is_visited: bool = False
    is_blocked: bool = False


class RouteSelectionState(BaseModel):
    selected_segment_id: str | None = None
    current_segment_id: str | None = None
    current_node: str | None = None
    target_node: str | None = None
    localization_source: str = "NONE"
    localization_confidence: str = "LOW"
    heading_hint_x: float | None = None
    heading_hint_y: float | None = None
    accepted: bool = False
    reason: str = ""


class RouteGuidanceDecision(BaseModel):
    accepted: bool = False
    reason: str = ""
    segment_id: str | None = None
    current_segment_id: str | None = None
    target_node: str | None = None
    confidence: float = 0.0
    heading_hint_x: float | None = None
    heading_hint_y: float | None = None
    safety_status: str = "UNKNOWN"


class SelectSegmentRequest(BaseModel):
    segment_id: str
    confirm_on_segment: bool = False


class HeadingHintRequest(BaseModel):
    segment_id: str
    heading_hint_x: float
    heading_hint_y: float

File: pc_app/web_ui/src/types/routeSegment.ts
export interface RouteSegment {
  segment_id: string;
  from_node: string;
  to_node: string;
  label: string;
  is_enabled: boolean;
  is_visited: boolean;
  is_blocked: boolean;
}

export interface RouteSelectionState {
  selected_segment_id: string | null;
  current_segment_id: string | null;
  current_node: string | null;
  target_node: string | null;
  localization_source: string;
  localization_confidence: string;
  heading_hint_x: number | null;
  heading_hint_y: number | null;
  accepted: boolean;
  reason: string;
}

export interface RouteGuidanceDecision {
  accepted: boolean;
  reason: string;
  segment_id: string | null;
  current_segment_id: string | null;
  target_node: string | null;
  confidence: number;
  heading_hint_x: number | null;
  heading_hint_y: number | null;
  safety_status: string;
}
